src/go/k8s/pkg/resources/configmap.go

```diff
@@ -53,27 +59,56 @@ type ConfigMapResource struct {
 	scheme		*runtime.Scheme
 	pandaCluster	*redpandav1alpha1.Cluster
 
-	svc	*ServiceResource
-	logger	logr.Logger
+	svc		*HeadlessServiceResource
+	logger		logr.Logger
+	nodePort	int32
 }
 
 // NewConfigMap creates ConfigMapResource
 func NewConfigMap(
 	client k8sclient.Client,
 	pandaCluster *redpandav1alpha1.Cluster,
 	scheme *runtime.Scheme,
-	svc *ServiceResource,
+	svc *HeadlessServiceResource,
 	logger logr.Logger,
 ) *ConfigMapResource {
 	return &ConfigMapResource{
-		client, scheme, pandaCluster, svc, logger.WithValues("Kind", configMapKind()),
+		client,
+		scheme,
+		pandaCluster,
+		svc,
+		logger.WithValues("Kind", configMapKind()),
+		0,
 	}
 }
 
 // Ensure will manage kubernetes v1.ConfigMap for redpanda.vectorized.io CR
-//nolint:dupl // we expect this to not be duplicated when more logic is added
 func (r *ConfigMapResource) Ensure(ctx context.Context) error {
 	var cfgm corev1.ConfigMap
+	IPsList := strings.Builder{}
+
+	if r.pandaCluster.Spec.ExternalConnectivity {
+		var nodeList corev1.NodeList
+
+		err := r.List(ctx, &nodeList)
+		if err != nil {
+			return fmt.Errorf("retrieving list of nodes: %w", err)
+		}
+
+		for i := range nodeList.Items {
+			externalIP, internalIP := GetNodeIPs(&nodeList.Items[i])
```

Comment:
so we're in dynamic environment, what happens when node list changes between us creating this configmap and launching redpanda? 

Also can we add some documentation as to why we need to do it this way - why this has to be in the configmap like this?

Reply:
I will add comment describing the situation.

Reply:
I need to admit. You are right that there is problem with this solution. As we don't watch for nodes changes the config map could be outdated when pod is scheduled to newly created node. I will reorganize the code to introduce list-watch functionality.